openghg/standardise/surface/_icos_corso.py

```diff
@@ -0,0 +1,465 @@
+    instrument: str,
+    inlet: str | None = None,
+    network: str = "ICOS",
+    sampling_period: str | None = None,
+    measurement_type: str | None = None,
+    update_mismatch: str = "never",
+    site_filepath: pathType | None = None,
+    **kwargs: dict,
+) -> dict:
+    """
+    Parse an icos files for the corso project
+
+    Args:
+        filepath: Path to file
+        site: Three letter site code
+        network: Network name
+        instrument: Instrument name
+        inlet: Optionally specify inlet height to check against filename
+        sampling_period: Sampling period e.g. 2 hour: 2H, 2 minute: 2m
+        measurement_type: Measurement type e.g. insitu, flask
+        header_type: ICOS data file with large (40 line) header or shorter single line header
+            Options: large, small
+        update_mismatch: This determines how mismatches between the internal data
+            "attributes" and the supplied / derived "metadata" are handled.
+            This includes the options:
+              - "never" - don't update mismatches and raise an AttrMismatchError
+              - "from_source" / "attributes" - update mismatches based on input data (e.g. data attributes)
+              - "from_definition" / "metadata" - update mismatches based on associated data (e.g. site_info.json)
+        site_filepath: Alternative site info file (see openghg/openghg_defs repository for format).
+            Otherwise will use the data stored within openghg_defs/data/site_info JSON file by default.
+    Returns:
+        dict: Dictionary of gas data
+    """
+
+    input_site = clean_string(site)
+    instrument = clean_string(instrument)
+    network = clean_string(network)
+    sampling_period = clean_string(sampling_period)
+    measurement_type = clean_string(measurement_type)
+
+    inlet = clean_string(inlet)
+    inlet = format_inlet(inlet)
+
+    if not isinstance(filepath, Path):
+        filepath = Path(filepath)
+
+    contributors = ["gns", "uheicrl", "noaa", "ucsd", "uheiiup"]
+    data_flag = None
+    columns_to_keep = None
+
+    # Currently contributors name included in the filename is the direct classifier as clean_14c_data. In future this may change and thus the contributors list will be required to be updated.
+    data_flag = ["clean_14c_data" for contributor_name in contributors if contributor_name in filepath.name]
+
+    species_fname = filepath.name.split(".")[-1]
+    species = synonyms(species_fname)
+
+    rename_dict = {
+        species_fname.lower(): species,
+        "samplingstart": "sampling_start",
+        "samplingend": "sampling_end",
+        "measunc": species + "_variability",
+        "estrep": species + "_repeatability",
+        "nbpoints": species + "_number_of_observations",
+        "stdev": species + "_variability",
+        "scallink": species + "_calibration_uncertainty",
+        "combunc": species + "_combined_uncertainty",
+        "integrationtime": species + "_sampling_period",
```

Comment:
Suggestion to change this back to what you had previously:
```suggestion
        "integrationtime": "integration_time",
```